### PR TITLE
Add explicit return statements in the pseudo-code

### PR DESCRIPTION
**Pull request by jedisct1**
This is especially useful for the Decrypt() function, to emphasize the fact that the function must return either an error, or `msg`.

### CODE REVIEW

**Comment by fscoto** [comment]
I was thinking about this, but the semantics quickly snowball. `throw` is undefined in the spec, especially since an error is supposed to be an output.

It now also looks like there are ordered returns.

Now that the error throw introduces `if`, I wonder if we shouldn't instead explicitly use an constant time comparison function we defined in the definitions section.

**Comment by jedisct1** [comment]
Maybe we can replace `throw` with `return a "Verification failed" error`?

The order or returned values doesn't matter. Would saying "return ct and tag" in plain English be better?

Introducing a constant-time comparison function is a good idea.

**Comment by fscoto** [comment]
Sounds good to me; please update accordingly and I have no reservations against merging

**Comment by samuel-lucas6** [comment]
> Maybe we can replace throw with return a "Verification failed" error?

Sounds best, although perhaps leave out the 'a' so it's `return "Verification failed" error`.

> The order or returned values doesn't matter. Would saying "return ct and tag" in plain English be better?

I think so.

> Introducing a constant-time comparison function is a good idea.

Agreed again.
